rust/src/pgsql/parser.rs

```diff
@@ -0,0 +1,2089 @@
+    pub length: u32,
+    pub message_body: Vec<PgsqlErrorNoticeMessageField>,
+}
+
+impl ErrorNoticeMessage {
+    pub fn new(identifier: u8, length: u32) -> Self {
+        ErrorNoticeMessage{
+            identifier,
+            length,
+            message_body: Vec::new(),
+        }
+    }
+}
+
+#[derive(Debug, PartialEq)]
+pub enum SSLResponseMessage {
+    SSLAccepted,
+    SSLRejected,
+    InvalidResponse,
+}
+
+impl From<u8> for SSLResponseMessage {
+    fn from(identifier: u8) -> Self {
+        match identifier {
+            b'S' => Self::SSLAccepted,
+            b'N' => Self::SSLRejected,
+            _ => Self::InvalidResponse,
+        }
+    }
+}
+
+impl From<char> for SSLResponseMessage {
+    fn from(identifier: char) -> Self {
+        match identifier {
+            'S' => Self::SSLAccepted,
+            'N'=> Self::SSLRejected,
+            _ => Self::InvalidResponse,
+        }
+    }
+}
+
+// TODO - do I really need this?
+impl fmt::Display for SSLResponseMessage {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            SSLResponseMessage::SSLAccepted => write!(f, "SSL Accepted"),
+            SSLResponseMessage::SSLRejected => write!(f, "SSL Rejected"),
+            _ => write!(f, "Invalid server response"),
+        }
+    }
+}
+
+#[derive(Debug, PartialEq)]
+pub struct ParameterStatusMessage {
+    pub identifier: u8,
+    pub length: u32,
+    pub param: PgsqlParameter,
+}
+
+#[derive(Debug, PartialEq)]
+pub struct BackendKeyDataMessage {
+    pub identifier: u8,
+    pub length: u32,
+    pub backend_pid: u32,
+    pub secret_key: u32,
+}
+
+#[derive(Debug, PartialEq)]
+pub struct ReadyForQueryMessage {
+    pub identifier: u8,
+    pub length: u32,
+    pub transaction_status: u8,
+}
+
+#[derive(Debug, PartialEq)]
+pub enum PgsqlBEMessage {
+    SSLResponse(SSLResponseMessage),
+    ErrorResponse(ErrorNoticeMessage),
+    NoticeResponse(ErrorNoticeMessage),
+    AuthenticationOk(AuthenticationMessage),
+    AuthenticationKerb5(AuthenticationMessage),
+    AuthenticationCleartextPassword(AuthenticationMessage),
+    AuthenticationMD5Password(AuthenticationMessage),
+    AuthenticationGSS(AuthenticationMessage),
+    AuthenticationSSPI(AuthenticationMessage),
+    AuthenticationGSSContinue(AuthenticationMessage),
+    AuthenticationSASL(AuthenticationSASLMechanismMessage),
+    AuthenticationSASLContinue(AuthenticationMessage),
+    AuthenticationSASLFinal(AuthenticationMessage),
+    ParameterStatus(ParameterStatusMessage),
+    BackendKeyData(BackendKeyDataMessage),
+    CommandComplete(RegularPacket),
+    ReadyForQuery(ReadyForQueryMessage),
+    RowDescription(RowDescriptionMessage),
+    DataRow(DataRowMessage),
+    DummyDataRow(RegularPacket),
+}
+
+impl fmt::Display for PgsqlBEMessage {
+    fn fmt(&self, f:&mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
+impl PgsqlBEMessage {
+    pub fn message_type_to_str(&self) -> &str {
```

Comment:
Just name this `to_str()`, I don't think there is any `to_str()` trait to worry about.